src/rewriter.coffee

```diff
@@ -512,6 +513,23 @@ exports.Rewriter = class Rewriter
         last_column:  prevLocationData.last_column
       return 1
 
+  # Add parens around a `do` IIFE followed by a chained `.` so that the
+  # chaining applies to the executed function rather than the function
+  # object (see #3736)
+  addParensToChainedDoIife: ->
+    condition = (token, i) ->
+      @tag(i - 1) is 'OUTDENT'
+    action = (token, i) ->
+      return unless token[0] is '.'
```

Comment:
Should be `token[0] in CALL_CLOSERS` to include `::` etc.

Reply:
@GeoffreyBooth updated to `in CALL_CLOSERS` and changed one test to use `?.` instead of `.`